errors/error.go

```diff
@@ -120,12 +143,55 @@ func ops(e *Error) []Op {
 // WrapGORMError wraps an GORM error into our error such as adding errors.Kind
 func WrapGORMError(op Op, err error) error {
 	if goErrors.Is(err, gorm.ErrRecordNotFound) {
-		return New(op, NotFound, err)
+		return New(nil, op, NotFound, err)
 	}
 
 	if goErrors.Is(err, gorm.ErrDuplicatedKey) {
-		return New(op, Conflict, err)
+		return New(nil, op, Conflict, err)
 	}
 
-	return New(op, err)
+	return New(nil, op, err)
+}
+
+// propagateContexts combines the "basics" and "extras" contexts from the child error into the parent, so that the
+// key-values propagate upwards to the top level error.
+func (e *Error) propagateContexts() {
+	subErr, ok := e.Err.(*Error)
+	if !ok {
+		return
+	}
+
+	subBasics, _ := subErr.ctx[ctxBasics].(common.Basics)
+	if subBasics != nil {
+		basics, basicsOK := e.ctx[ctxBasics].(common.Basics)
+		if !basicsOK {
+			basics = common.Basics{}
+		}
+
+		collection.Copy(basics, subBasics)
+		e.ctx[ctxBasics] = basics
+	}
+
+	subExtras, _ := subErr.ctx[ctxExtras].(common.Extras)
+	if subExtras != nil {
+		extras, extrasOK := e.ctx[ctxExtras].(common.Extras)
+		if !extrasOK {
+			extras = common.Extras{}
+		}
+
+		collection.Copy(extras, subExtras)
+		e.ctx[ctxExtras] = extras
+	}
+
+	subErr.ctx = nil
+}
+
+func (e *Error) getBasics() common.Basics {
```

Comment:
Should we use `basics()` instead?
Ref: https://google.github.io/styleguide/go/decisions#getters

Reply:
Renamed to `basics()` and `extras()` in 75b6d5e.